Auth0Tests/JWKSpec.swift

```diff
@@ -0,0 +1,82 @@
+// JWKSpec.swift
+//
+// Copyright (c) 2020 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import Quick
+import Nimble
+import OHHTTPStubs
+
+@testable import Auth0
+
+class JWKSpec: QuickSpec {
+    
+    override func spec() {
+        
+        describe("public key generation") {
+            
+            let jwk = generateRSAJWK(from: TestKeys.rsaPublic)
+            
+            if #available(iOS 10, OSX 10.12, tvOS 10, watchOS 3, *) {
+                context("successful generation") {
+                    it("should generate a RSA public key") {
+                        let publicKey = jwk.rsaPublicKey!
+                        let keyAttributes = SecKeyCopyAttributes(publicKey) as! [String: Any]
+                        
+                        expect(keyAttributes[String(kSecAttrKeyType)] as? String).to(equal(String(kSecAttrKeyTypeRSA)))
+                    }
+                }
+            }
+            
+            context("unsuccessful generation") {
```

Comment:
are exponent and modulus the only reasons why it could fail? What about having a different usage or algorithm? I don't expect all the cases to be covered, but those 2 I'd say are a must

Reply:
The usage is not used at all in the `.rsaPublicKey` implementation. All that's needed is valid values in the properties `rsaModulus` and `rsaExponent`.

Reply:
I can add a check in the guard clause of `.rsaPublicKey ` to return nil if the algorithm != "RSA" and usage != "sig".

Reply:
After all, it's an internal (package-private) property.

Reply:
Done in 123af59498094926539a355378c5ce78efa7a1bb. Thanks for spotting it!

Reply:
@Widcket for clarification. The `use` key is optional according to the spec, but when present it should be used to check whether the key you are reading can be used for the operation you want to perform. 

For this particular implementation, it means that if `use!='sig'` then the key is invalid and should not be used.

Reply:
Then I should update the implementation, because I added a check in the guard clause testing if usage == "sig". I'll need to split it into a separate check, because `use` is optional, and only if present should be part of the validation. Thanks for the clarification.